### ISSUE TITLE
Higher GPU consumption for Lenet-5 and LSTM models when compared to other frameworks

### ISSUE THREAD

**Issue by jf9123** [pytorch/pytorch#100425]
### 🐛 Describe the bug

Hello everybody.

I’ve been experimenting with different models and different frameworks, and I’ve noticed that, when using GPU, training Lenet-5 and LSTM models on the IMDB dataset and the MNIST, respectively, leads to higher GPU usage to the Keras and TensorFlow v1.X implementations. Moreover, I’ve also noticed that higher GPU consumption leads to faster training and inference times, and improved accuracy for the Lenet-5 model, while it leads to a slower training time, faster inference time, and consistent accuracy, when compared to the other frameworks

I'm on a Ubuntu 18.04.4 system equipped with an NVIDIA Quadro RTX 4000 GPU with 8GB of VRAM and an Intel(R) Core(TM) i9-9900K CPU running at 3.60GHz.

### Versions

The version of relevant libraries are:

numpy==1.19.5
torch==1.10.0
torchaudio==0.10.0
torchvision==0.11.1
mkl==2022.2.1
mkl-fft==1.3.0
mkl-random==1.2.1
mkl-service==2.4.0

**Comment by jf9123** [comment]
![gpu cons](https://user-images.githubusercontent.com/132307143/235548546-7b9aeb70-1408-4b78-a3ca-fc3f63d8f21e.png)
 
Here are boxplots that showcase the performance. Note that the blue boxplots represent training time while the orange ones represent the inference time.

**Comment by janeyx99** [comment]
@ngimel Do we have an owner for memory usage? Are these results expected?
